### DISCUSSION TITLE
Extending DTLS with Connection ID

### DISCUSSION THREAD

**Discussion by YSLew-tado** [openthread/openthread#9363] [Ideas]
Hi,

I noticed that [MbedTLS 3.3.0](https://github.com/Mbed-TLS/mbedtls/releases/tag/v3.3.0) adds support for DTLS Connection ID.
It would be great if this could be leveraged to extend OpenThread DTLS.

For my simple use case of client being prepared to send using a CID but does not wish the server to use one when sending, it seems relatively easy to do. [enable_cid.patch.txt](https://github.com/openthread/openthread/files/12386842/enable_cid.patch.txt)
[Openthread-CID.pcapng.txt](https://github.com/openthread/openthread/files/12386845/Openthread-CID.pcapng.txt)

I invite everyone more familiar with the codebase to share their opinion whether this is a good idea or maybe there is a simpler / cleaner way.

Thanks

**Comment by jwhui** [comment]
I think it makes sense to allow support for DTLS Connection IDs. We should probably wrap it with the `MBEDTLS_SSL_DTLS_CONNECTION_ID` build flag.
